ui/raidboss/data/07-dt/raid/r4n.ts

```diff
@@ -1,14 +1,274 @@
+      type: 'HeadMarker',
+      netRegex: { id: headMarkerData.multiHitStack, capture: true },
+      response: Responses.stackMarkerOn(),
+    },
+    {
+      id: 'R4N Headmarker Thunderstorm Spread',
+      type: 'HeadMarker',
+      netRegex: { id: headMarkerData.spread, capture: false },
+      suppressSeconds: 5,
+      response: Responses.spread(),
+    },
+    {
+      id: 'R4N Headmarker Wicked Jolt Tankbuster',
+      type: 'HeadMarker',
+      netRegex: { id: headMarkerData.tankBusterLine, capture: true },
+      response: Responses.tankBuster(),
+    },
+    {
+      id: 'R4N Wrath of Zeus',
+      type: 'StartsUsing',
+      netRegex: { id: '92C7', source: 'Wicked Thunder', capture: false },
+      response: Responses.aoe(),
+    },
+    {
+      id: 'R4N Sidewise Spark Go Left',
+      type: 'StartsUsing',
+      netRegex: { id: ['92BC', '92BE'], source: 'Wicked Thunder', capture: false },
+      response: Responses.goLeft(),
+    },
+    {
+      id: 'R4N Sidewise Spark Go Right',
+      type: 'StartsUsing',
+      netRegex: { id: ['92BD', '92BF'], source: 'Wicked Thunder', capture: false },
+      response: Responses.goRight(),
+    },
+    {
+      id: 'R4N Left Roll',
+      type: 'Ability',
+      netRegex: { id: '92AC', source: 'Wicked Thunder', capture: false },
+      response: Responses.goWest(),
+    },
+    {
+      id: 'R4N Right Roll',
+      type: 'Ability',
+      netRegex: { id: '92AB', source: 'Wicked Thunder', capture: false },
+      response: Responses.goEast(),
+    },
+    {
+      id: 'R4N Threefold Blast Initializer',
+      type: 'StartsUsing',
+      netRegex: { id: ['92AD', '92B0'], source: 'Wicked Thunder', capture: false },
+      run: (data) => data.expectedBlasts = 3,
+    },
+    {
+      id: 'R4N Fourfold Blast Initializer',
+      type: 'StartsUsing',
+      netRegex: { id: ['9B4F', '9B55'], source: 'Wicked Thunder', capture: false },
+      run: (data) => data.expectedBlasts = 4,
+    },
+    {
+      id: 'R4N Fivefold Blast Initializer',
+      type: 'StartsUsing',
+      netRegex: { id: ['9B56', '9B57'], source: 'Wicked Thunder', capture: false },
+      run: (data) => data.expectedBlasts = 5,
+    },
+    {
+      id: 'R4N XFold Blast Collector',
+      type: 'GainsEffect',
+      netRegex: { effectId: 'B9A', count: Object.values(effectB9AMap), capture: true },
+      condition: (data, matches) => {
+        const count = matches.count;
+
+        if (!isEffectB9AValue(count))
+          return false;
+        data.storedBlasts.push(count);
+
+        return data.storedBlasts.length >= data.expectedBlasts;
+      },
+      durationSeconds: (data) => {
+        if (data.expectedBlasts === 3)
+          return 14.4;
+        if (data.expectedBlasts === 4)
+          return 18.9;
+        return 23.2;
+      },
+      infoText: (data, _matches, output) => {
+        const expectedBlasts = data.expectedBlasts;
+        data.expectedBlasts = 3;
```

Comment:
Yes, if it's possible I think (and this is opinion only) it's slightly better to avoid writing to the `data` object from within popup alert functions. Sometimes it's unavoidable, but it's easier to reason about things later on if we can be confident that any changes to `data` are in `run` blocks.